internal/blocksync/pool.go

```diff
@@ -796,12 +794,14 @@ OUTER_LOOP:
 				}
 				return
 			case <-bpr.Quit():
+				retryTimer.Stop()
 				return
 			case <-retryTimer.C:
 				if !gotBlock {
 					bpr.Logger.Debug("Retrying block request(s) after timeout", "height", bpr.height, "peer", bpr.peerID, "secondPeerID", bpr.secondPeerID)
 					bpr.reset(bpr.peerID)
 					bpr.reset(bpr.secondPeerID)
+					retryTimer.Stop()
```

Comment:
What's the point of stopping the timer when it's already fired?

Reply:
thank you, my misstake